### PR TITLE
change mapbox library on manage projects  page

### PR DESCRIPTION
**Pull request by sarveshpro**
Changes in this pull request:
- Add project sites map using `react-mapbox-gl` and `react-mapbox-gl-draw`
- support editing uploaded polygon
- overall better polygon drawing exp.

challenges in this PR - onclick handler on project details location map goes into a loop.

### CODE REVIEW

**Comment by norbertschuler** [comment]
I add the polygon on a normal map view:

Bildschirmfoto 2020-11-27 um 16 32 11

But I see them afterwards on a satellite view:

Bildschirmfoto 2020-11-27 um 16 32 50

I that wanted?

File: package.json [hunk 1]
```diff
@@ -52,11 +53,13 @@
     "dom-to-image": "^2.6.0",
     "express": "^4.17.1",
     "framer-motion": "^2.9.4",
+    "geojson-validation": "^1.0.2",
     "i18next": "^19.8.3",
     "i18next-browser-languagedetector": "^6.0.1",
     "layer-manager": "^3.0.8",
     "luma.gl": "^7.3.2",
     "mapbox-gl": "^1.12.0",
+    "mapbox-gl-draw": "^0.16.0",
```
Comment (norbertschuler) [inline package.json:62]: Are the modules still used?

Comment (norbertschuler) [reply package.json:62]: Found usage of `mapbox-gl` at https://github.com/Plant-for-the-Planet-org/planet-webapp/blob/3aec7a3f96da4c32d0e47fd902c2b7db0179a50b/pages/_app.tsx#L2 but no usage of `mapbox-gl-draw`, right?

File: package.json [hunk 2]
```diff
@@ -74,6 +77,8 @@
     "react-lazyload": "^3.1.0",
     "react-map-gl": "^5.2.8",
     "react-map-gl-draw": "^0.21.1",
```
Comment (norbertschuler) [inline package.json:79]: Also could not find usage of `react-map-gl-draw` anymore, right?

**Review by norbertschuler — CHANGES_REQUESTED** [review]
Maybe you can clean up unused modules (I do not understand why we use so much different modules for maps.)

File: package.json (hunk 1 above)
Comment (sarveshpro) [reply package.json:62]: `mapbox-gl-draw` is a required dependency of `react-mapbox-gl-draw` https://www.npmjs.com/package/react-mapbox-gl-draw
react-mapbox-gl-draw is just a react wrapper for the `mapbox-gl-draw` original mapbox library.

File: package.json (hunk 2 above)
Comment (sarveshpro) [reply package.json:80]: Yes removing that. we don't need anymore. `react-mapbox-gl-draw` replaces it.

**Review by harshvitra — APPROVED** [review]
LGTM

**Review by norbertschuler — APPROVED** [review]

**Pull request merged**